.github/workflows/gh-pages.yml

```diff
@@ -28,27 +28,39 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
+      - name: Cache Setup
+        id: cache-ocaml-setup
+        uses: actions/cache@v3
+        with:
+          path: ~/.opam
+          key: ${{ runner.os }}-ocaml-setup
+
       # sphinx-multiversion uses tags to build
       # the documentation in multiple versions.
       - name: Checkout | Fetch tags
         run: git fetch --tags origin
 
       - name: Setup | mdBook | 1/2
         uses: hecrj/setup-rust-action@master
-      
+
       - name: Setup | mdBook | 2/2
         uses: peaceiris/actions-mdbook@v1
         with:
-          mdbook-version: 'latest'
+          mdbook-version: "latest"
 
       - name: Setup | Update
         run: sudo apt-get update
 
-      - name: Setup | Doxygen
-        run: sudo apt-get install doxygen
+      - name: Setup | Ubuntu
```

Comment:
```diff
-Setup | Ubuntu
+Setup | Tools
```

Reply:
I used system if that's better. It's not setting up tools. It's just installing system packages.